pkg/gameservers/controller.go

```diff
@@ -639,6 +639,50 @@ func (c *Controller) addGameServerHealthCheck(gs *agonesv1.GameServer, pod *core
 	})
 }
 
+func (c *Controller) addSDKEnvironmentVarsToGameServer(gs *agonesv1.GameServer, pod *corev1.Pod) {
```

Comment:
I think we need to set the environment variables to all containers, not just the gameserver one.

Reason being, lots of people are running sidecars with the sdk embedded there, rather than doing the direct integration into the gameserver binary itself.

So we need to ensure the SDK will work in any container that is running in the GameServer Pod

Reply:
Makes sense. I've reversed the logic to apply to all containers except for the sdksidecar instead of just the indicated game server container.